lib/sm/client.rb

```diff
@@ -171,12 +177,19 @@ def get_folder_messages(user_uuid, folder_id, use_cache)
     #
     # @return [Common::Collection]
     #
-    def get_folder_threads(folder_id, page_size, page_number, sort_field, sort_order)
-      path = "folder/threadlistview/#{folder_id}"
+    def get_folder_threads(params)
+      requires_oh_messages = params[:requires_oh_messages]
+      base_path = "folder/threadlistview/#{params[:folder_id]}"
+      query_params = [
+        "pageSize=#{params[:page_size]}",
+        "pageNumber=#{params[:page_number]}",
+        "sortField=#{params[:sort_field]}",
+        "sortOrder=#{params[:sort_order]}"
```

Comment:
Are these params required? what if they don't exist?

Reply:
great question. `pageSize`, `pageNumber`, `sortField`, `sortOrder` are optional. But `folder_id` is required. I may need to adjust for that

Reply:
adding `folder_id` as a separate argument